apps/prairielearn/src/lib/socket-server.js

```diff
@@ -1,67 +1,58 @@
 // @ts-check
 const { Server } = require('socket.io');
-const redis = require('redis');
+const { Redis } = require('ioredis');
 const { createAdapter } = require('@prairielearn/socket.io-redis-adapter');
 const { logger } = require('@prairielearn/logger');
 const path = require('path');
 const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
 
 const { config } = require('./config');
 
+/**
+ * @param {Redis} client
+ * @param {string} type
+ */
+function attachEventListeners(client, type) {
+  client.on('error', (err) => {
+    logger.error(`redis ${type} client: error`, err);
+  });
+  client.on('connect', () => {
+    logger.verbose(`redis ${type} client: connect`);
+  });
+  client.on('ready', () => {
+    logger.verbose(`redis ${type} client: ready`);
+  });
+  client.on('reconnecting', () => {
+    logger.verbose(`redis ${type} client: reconnecting`);
+  });
+  client.on('close', () => {
+    logger.verbose(`redis ${type} client: close`);
+  });
+  client.on('end', () => {
+    logger.verbose(`redis ${type} client: end`);
+  });
```

Comment:
There is also the [`select`](https://github.com/redis/ioredis#connection-events) event, which isn't relevant for us right now (we aren't using Redis cluster) but might be worth including now just for completeness. I guess `wait` is also in this category -- maybe we should add it now just in case we change things in the future so we don't miss events.